cluster-operator/src/main/java/io/strimzi/operator/cluster/model/VolumeUtils.java

```diff
@@ -257,4 +284,80 @@ public static VolumeMount createVolumeMount(String name, String path) {
     public static String getVolumePrefix(Integer id) {
         return id == null ? AbstractModel.VOLUME_NAME : AbstractModel.VOLUME_NAME + "-" + id;
     }
+
+    /**
+     * Volume names have to follow DNS label standard form RFC1123:
+     *     - contain at most 63 characters
+     *     - contain only lowercase alphanumeric characters or ‘-’
+     *     - start with an alphanumeric character
+     *     - end with an alphanumeric character
+     *
+     *  This method checkes if the volume name is a valid name and if not it will modify it to make it valid.
+     *
+     * @param originalName  The original name of the volume
+     * @return              Either the original volume name or a modified version to match volume name criteria
+     */
+    public static String getValidVolumeName(String originalName) {
+        if (originalName == null) {
+            throw new RuntimeException("Volume name cannot be null");
+        }
```

Comment:
`Objects.requireNonNull(originalName, "Volume name cannot be null")` ?

Reply:
Do I have to? :-o TBH I really find traditional if more readable. And from a quck search it doesn't seem like we are using `Objects.requireNonNull` everywhere apart from this part of code but rather as an exception.

Reply:
No, you don't have to.

I raised it as a question because it seems to be becoming an idiom in other projects (including vertx, for example). It's nice in that it's a one-liner, but I don't particularly like that it throws an NPE, because in cases like this, where it's asserting a precondition, I think IAE would be better. But really all I want is for _us_ to have a consensus about whether we adopt this idiom, and if we decide to not use `requireNonNull()` then I do think we should be consistent about the exception we throw. I also think we could be a little more strict about checking preconditions, but that's irrelevant to this PR.